Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -358,27 +358,38 @@ PhysicalParticleContainer::AddPlasmaFromFile(ParticleReal q_tot,
             weight = w * w_unit;
         }
 
+        // FIXME: Only injects particles on coarsest grid, so changes are needed to allow injection to finer grid levels.
+        const Geometry& geom = Geom(0);
+        const auto problo = geom.ProbLoArray();
+        const auto probhi = geom.ProbHiArray();
+
         for (auto i = decltype(npart){0}; i<npart; ++i){
             ParticleReal const x = ptr_x.get()[i]*position_unit_x;
+            bool const in_box_x = x>problo[0] && x<probhi[0];
+            if( not in_box_x ) continue;
             ParticleReal const z = ptr_z.get()[i]*position_unit_z+z_shift;
 #   ifndef WARPX_DIM_3D
+            bool const in_box_z = z>problo[1] && z<probhi[1];
+            if( not in_box_z ) continue;
             ParticleReal const y = 0.0_prt;
+            ParticleReal const uy = 0.0_prt;
 #   else
+            bool const in_box_z = z>problo[2] && z<probhi[2];
+            if( not in_box_z ) continue;
             ParticleReal const y = ptr_y.get()[i]*position_unit_y;
+            bool const in_box_y = y>problo[1] && y<probhi[1];
+            if( not in_box_y ) continue;
+            ParticleReal const uy = ptr_uy.get()[i]*momentum_unit_y/PhysConst::m_e;
 #   endif
-            if (plasma_injector->insideBounds(x, y, z)) {
```

Comment:
I have the feeling that the optimization of reads is not too relevant and we could just try to fix this call here.

In fact, it looks like these bounds are set to inf by default (instead of the sim box) and that user-parameters are already present:
https://github.com/ECP-WarpX/WarpX/blob/master/Source/Initialization/PlasmaInjector.cpp#L91-L96

Reply:
Thanks, will change `plasma_injector` default to box so we can make the fix of the current PR applicable to all the injection schemes.
And will update documentation on this too.